src/pymanopt/function.py

```diff
@@ -70,12 +77,17 @@ def inner(cost: Callable[..., Any]) -> Function:
                 "Decorated function must only accept positional arguments "
                 "or a variable-length argument like *x"
             )
-            backend = getattr(
+            backend_type = getattr(
                 import_module(
                     f"pymanopt.backends.{module}",
                 ),
                 backend_class,
-            )()
+            )
+            backend = (
+                backend_type(dtype=dtype)
+                if dtype is not None
+                else backend_type()
```

Comment:
comment that using the real version doesn't change anything for a function because the autodiff methods are the same